src/index.js

```diff
@@ -1,58 +1,36 @@
-import React, { useRef, useEffect, useState, memo } from "react";
+import React, { memo, useCallback, useState } from "react";
 import PropTypes from "prop-types";
-import Lottie from "lottie-react-web";
+import Lottie from "react-lottie-player";
 import parseUnit from "parse-unit";
 
-const options = Object.freeze({
-  animationData: require("./animationData.json"),
-  autoplay: false,
-  loop: true
-});
+const animationData = require("./animationData.json");
 
-const NightModeToggle = ({ size, checked, onChange, speed, className, ...extraProps }) => {
-  const ref = useRef();
-  const [progress, setProgress] = useState(() => 0);
+const NightModeToggle = ({ size, checked, onChange, speed, className }) => {
   const [sizeValue, sizeUnit] = parseUnit(size);
-  const segments = checked ? { segments: [2, 96] } : null;
-  useEffect(() => {
-    if (progress >= 0.5) {
-      if (checked) {
-        ref.current.anim.pause();
-      } else if (ref.current.anim.isPaused) {
-        ref.current.anim.play();
-      }
-    } else if (!checked) {
-      ref.current.anim.pause();
-    }
-  }, [checked, progress]);
-  useEffect(
-    () => {
-      /* force */
-      (!!checked) && ref.current.anim.setCurrentRawFrameValue(48);
-    },
-    [],
-  );
-  const [eventListeners] = useState(() => [
-    {
-      eventName: "enterFrame",
-      callback: ({ currentTime, totalTime }) =>
-        setProgress(currentTime / totalTime)
-    }
-  ]);
+  const [isReadyToAnimate, setReadyToAnimate] = useState(false);
```

Comment:
We use this state, along with the `segementToJumpToImmediately` variable below to ensure that the underlying lottie animation is *not* executed when the component first mounts.  After the first click this stays true forever.

Reply:
Without this logic the component will start in the opposite position and then animate to the correct state, which is not desirable.